dimod/reference/composites/connectedcomponent.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2019 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# =============================================================================
+"""
+A composite that break the problem into sub-problems corresponding to the connected components of the binary
+quadratic model graph before sending to its child sampler.
+"""
+
+from dimod.core.composite import ComposedSampler
+from dimod.sampleset import SampleSet
+import networkx as nx
```

Comment:
One problem is that dimod actually does not have a dependency on networkx, it's optional. Because bfs/dfs is pretty easy to implement, it might be better to not go through networkx to find the connected components. I could add a dimod version of `nx.connected_components` pretty easily.

Reply:
Once I have `dimod.connected_components`, I can modify the code accordingly.